k9mail-library/src/test/java/com/fsck/k9/mail/filter/FixedLengthInputStreamTest.java

```diff
@@ -0,0 +1,230 @@
+                "MEzQrUETaHqkknZFwBKQJJwgZaceaXFmEBLKBoCzcTXayTIEOlGngzfzvkSrDmoOcZroyQSyukxxbJdlaEzDEDJOaSIszGkHeomY" +
+                "KipWsMrKHIYpXaqvstsBlaFRhHyvuwtCdYsjWBtVPDNwTtOuPYklKIHAkvoDceSqonrwWtORQrcZKpiNKzFLqbWgfmnhCcKCeXng" +
+                "uexijkkVShZXIvcYgZdljYyIOQCrlLtHcGMZNJFhCcWJqcriZLRZctrrESxxKZWkGeVcKLFMTHcMwidbiAchSpjQBvCwMXpAlpSt" +
+                "seWUSbdfGQqdadWczKVrEYaiDrhNeepPFDllynREhXUBspdOsRPkVMVaNbcmJEwQkoMeWlsfvRpdAuBZYTzfKVDKQGDoDVDhJyeA" +
+                "BnzjGEEoiGrOiLGoViHpiLCBOLBiLCtfxSWKYuKuixEAJOBiMZFbHGBVDJmUNoeXfxXKNabolkDTCwcPohDrsvvinNCHDSOYnfXp" +
+                "olFRYTClufMizjJMzgBYApgrhOFZpAmmJaTadOgIVVafWvEFKOefUILvQNLaxdWuDVmzOxsSIZiWvYWbpfpIqEMgmfaUpHVdxHOu" +
+                "xbYPqezLbQPjXwULNznEyHYpyCpAFOawJpMmHfluqsSNMzNVlyafSPLDywjvEMIqcYWSmDdcfUQMHbzkECgkrYJuinAWYUPjoJRZ" +
+                "QkxcfkTlhhoFnmbZyvLSrIVcNYdLIDUsqCLQAXEJIJVegUJPfzssOmWHDAPZyLtylUUgFyxvwxLFgAnehydWWXnXIVmnqZEeXeAh" +
+                "cANnXpHqMcAtNXgmBkELyqrYemyPsnlvWMNFJsjJcYNoaAwFiAKVxebtwLLvodQFtuJxasxTcQvsiwSRjWanjWnMDoJKoQtwOcCK" +
+                "mKxHxyUuggufOjgJGvUoUooMBpMXDGrNohNzJbQmnCuFVHxulXEircCYtezkVeEkJKFRDXUlwvOSrfhTmnoytjeQbBcIaRZYsRAn" +
+                "isNlMSHqEXQIDirVGicrcOUxQzYhZlnHqwbKslxGkAjIPesZeQKSgbDMnFaAmVLOOrONYuszgMCNCvNGYJFOWGpZUmNvWvTWFoEX" +
+                "QrPSvEmhxyYmFuIfJppvJSDlggOwcXYiXdtgoFfRZyTQbQmdnhoVHvQffkcMSqHDGRpwSLRbqDWwHJOkVJSMFzHdttJwyVpWuTDa" +
+                "uFXOuHfpoXkHStuKkreyWLsasXTRVchqRaEYjUmcTuNZYlcVPFezKqIaDiRcRxuuYFtMzJqOIwMNmITofhdQiXTIfUHGfsYnPYgM" +
+                "wpVdzGoyIrSSKJjWMnGdhmJHwHoFCqNtXOnbltOMnlCuzBsNUIlrKgfHRTAzAoFAlMvxgIrGffqCLtpAdbCbxzxMuivOctNDhuXy" +
+                "iTuSYUvHnxZxklJDuMbMeGqQddtmAMVAhGNPVfdBaNQiDNxrRqLqFuYMjkDwekzDSgkFbCEiHPwmAqAuoRDrtnwXQfkhyKLNlhRE" +
+                "ZxDybKsCrrQPcrNFKskrvceHPGjvyBtVgaHTSmtzdMSVwNyMIlmsMIgZqpVCNVBmwPVYkEfuenVdGeQcSdGSEMuJGOdWXyMbVQiN" +
+                "hKKwnoffMoKJEbfgSLOMsSVUwEBRGFBPFhhsKLZrmPIdSMKwxhPOvzGywVCamswfIspSUHXyvtQrWGNUkyVyYWwjwumRSVfDFepT" +
+                "JnNHLfbAdkCfHsEbUQCHBlSlgpYExOjYyuYbLjCjBYdNXcfdShRRGhzcdsxmsbAnHunqzIeVsOrlJRfdhaDYtmeIOUBCqFZBpLXT" +
+                "jkvAhXYTjLCmcKebgCUFWCLgYnfxYIrTnbLuEyNPeIlyNIWefQrpYKHtEaUVESWrHYFWGHObaKnpspDilNhkfrWdFXUyZEgkbBPC" +
+                "DSUAsUKPpFNyrWPZCPHzVARrspCJbXNCkotJGtgrDdLMMGIPLEPirRZzRxYgDiefAsUvCTnQXGHTOXTCRgjZCegPBxXTblJVFepV" +
+                "YzPbwggNJDRgrHbtLvEQilSMnHSmeJahMUjJXXHFwZXQlMRVrPolpogKooCqIYErEiNnaWrbOaQhLRVxGCGHVnSaHyqaFQwdaSUF" +
+                "JRjJWyDvRjwIAVYTEghKdtdHnATaBxrExrpBmIdbZciYLtZszlxmziScLwdJWaXVycSAdubKkZGlozHnjBVRqllIzFufoSLgxExE";
+
+        STR_LENGTH = string.length();
+    }
+
+    // Test the cases that fixed length is smaller than the length of bytes array
+
+    @Test
+    public void testLongInput() throws IOException {
```

Comment:
Out tests generally follow the "arrange - act - assert" pattern. Each stage separated by an empty line.
Rewritten with okio this test could look like this:

```java
@Test
public void testLongInput() throws IOException {
    InputStream inputStream = new Buffer().writeUtf8(string).inputStream();
    FixedLengthInputStream fixedLengthInputStream = new FixedLengthInputStream(inputStream, LENGTH_LIMIT1);

    String readFromStream = readString(fixedLengthInputStream);

    String expected = string.substring(0, LENGTH_LIMIT1);
    assertEquals(expected, readFromStream);
}

private String readString(InputStream inputStream) throws IOException {
    return Okio.buffer(Okio.source(inputStream)).readUtf8();
}
```